src/sync.rs

```diff
@@ -1,15 +1,38 @@
+#[cfg(all(not(feature = "std"), loom))]
+compile_error!("loom requires the standard library");
+
+//loom support
+
 #[cfg(loom)]
 pub(crate) mod atomic {
-    pub(crate) use loom::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use loom::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
 #[cfg(loom)]
 pub(crate) use loom::thread::yield_now;
 
-#[cfg(not(loom))]
+//std imports
+
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) mod atomic {
-    pub(crate) use std::sync::atomic::{fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize};
+    pub(crate) use std::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
 }
-
-#[cfg(not(loom))]
+#[cfg(all(not(loom), feature = "std"))]
 pub(crate) use std::thread::yield_now;
+
+//core imports
+
+#[cfg(all(not(loom), not(feature = "std")))]
+pub(crate) mod atomic {
+    pub(crate) use core::sync::atomic::{
+        fence, AtomicIsize, AtomicPtr, AtomicU64, AtomicUsize, Ordering,
+    };
+}
+
+#[cfg(all(not(loom), not(feature = "std")))]
+pub(crate) fn yield_now() {
+    //FIXME: What goes here?
```

Comment:
Maybe [`hint::spin_loop`](https://doc.rust-lang.org/std/hint/fn.spin_loop.html)? Though I'd rather run that directly where this is called than "hide" it between the name `yield_now`.